### PR TITLE
Streamline and update README.md

### PR DESCRIPTION
**Pull request by anvega**
As discussed during the last leads meeting, this PR updates the readme to make it more streamlined and less bloated. Changes include:

- Combined the Objective, Background, and Mission sections under a single "About Us" header for a more cohesive introduction.
- Simplified language and content throughout the document for clarity and ease of understanding.
- Added a table format for publications to improve readability and accessibility, sorted by date in ascending order.
- Updated meeting information to be less words but keep direct links to CNCF and TAG Security calendars.

These updates aim to provide a more concise and friendly experience for visitors and contributors.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,120 +2,87 @@
...
-Got something to bring up or share? Review how to get a topic or presentation added to the Agenda on our [process](governance/process.md#getting-on-the-agenda) page.
+To add a topic to the agenda, review our [process](governance/process.md#getting-on-the-agenda).
 
 ## Gatherings
 
-Please let us know if you are going and if you are interested in attending (or
-helping to organize!) a gathering. Create
-a [github issue](https://github.com/cncf/tag-security/issues/new) for an event
-and add to list below:
+Interested in attending or organizing an event? Create a [GitHub issue](https://github.com/cncf/tag-security/issues/new):
```
Comment (eddie-knight) [inline README.md:73]: Why an issue for attending an event?

Comment (anvega) [reply README.md:73]: Cargo cult. Will remove.

Comment (anvega) [reply README.md]: After further consideration, I've noticed that people typically don't initiate events by filing an issue. Instead, they organize and plan events independently through platforms like Luma or Meetup, or through KCD. Therefore, I removed the suggestion to file a GitHub issue. If anyone is organizing an event, they can handle it themselves and announce it on social media, but it won't be considered our official event. I have retained SecurityCon and past events in the list.

**Review by JustinCappos — APPROVED** [review]
PR looks good overall.  The assessments are fairly numerous, so it may or may not be wise to move them to a different section or de-emphasize them somehow.  I'm fine with how it is now though and will defer to others.

**Comment by eddie-knight** [comment]
I agree with @JustinCappos that a list of publications (especially assessments) is cumbersome on the front page.

**Comment by anvega** [comment]
One of the key principles of intentional Information Architecture is prioritizing the visitor's experience. We need to consider what visitors are looking for and how they can easily find that information, whether through search or navigation.

If someone has heard about an assessment and comes here to find it, having it readily available on the front page allows them to locate it quickly without needing to click through multiple pages or dig for it. More cumbersome is coming to find something quickly and struggling to locate it when it's buried three levels deep. 

For visitors who are unaware of the assessments, having a selection of key ones listed on the front page serves as an introduction, showing that these resources exist. Perhaps a good approach is to specify that the listed assessments are selected samples, and provide a note that the complete list can be found in the directory.

**Comment by anvega** [comment]
I unlisted the assessments from the publication list based on feedback.

**Comment by eddie-knight** [comment]
Thank you @anvega!

**Pull request merged**